modules/material-management/test/cryptographic_material.test.ts

```diff
@@ -218,80 +173,21 @@ describe('decorateCryptographicMaterial', () => {
     const trace = {
       keyNamespace: 'k',
       keyName: 'k',
-      flags: KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY,
     }
     expect(() => test.setUnencryptedDataKey(dataKey, trace)).to.throw(
       'Unencrypted Master Key must be an isolated buffer.'
     )
   })
 
-  it('Precondition: Trace must be set, and the flag must indicate that the data key was generated.', () => {
-    const suite = new NodeAlgorithmSuite(
-      AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16
-    )
-    const test = decorateCryptographicMaterial(
-      { suite, keyringTrace: [] } as any,
-      KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
-    )
-    const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
-    expect(() =>
-      test.setUnencryptedDataKey(new Uint8Array(dataKey), {} as any)
-    ).to.throw('Malformed KeyringTrace')
-  })
-
-  it('Precondition: On set the required KeyringTraceFlag must be set.', () => {
-    const suite = new NodeAlgorithmSuite(
-      AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16
-    )
-    const test = decorateCryptographicMaterial(
-      { suite, keyringTrace: [] } as any,
-      KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
-    )
-    const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
-    const trace = {
-      keyNamespace: 'k',
-      keyName: 'k',
-      flags: KeyringTraceFlag.WRAPPING_KEY_SIGNED_ENC_CTX,
-    }
-    expect(() =>
-      test.setUnencryptedDataKey(new Uint8Array(dataKey), trace)
-    ).to.throw('Required KeyringTraceFlag not set')
-  })
-
-  it('Precondition: Only valid flags are allowed.', () => {
-    const suite = new NodeAlgorithmSuite(
-      AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16
-    )
-    const test = decorateCryptographicMaterial(
-      { suite, keyringTrace: [] } as any,
-      KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
-    )
-    const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
-    const trace = {
-      keyNamespace: 'k',
-      keyName: 'k',
-      flags:
-        KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY |
-        KeyringTraceFlag.WRAPPING_KEY_DECRYPTED_DATA_KEY,
-    }
-    expect(() =>
-      test.setUnencryptedDataKey(new Uint8Array(dataKey), trace)
-    ).to.throw('Invalid KeyringTraceFlags set.')
-  })
-
   it('Precondition: The unencryptedDataKey must not have been modified.', () => {
     const suite = new NodeAlgorithmSuite(
       AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16
     )
-    const material = decorateCryptographicMaterial(
-      { suite, keyringTrace: [] } as any,
-      KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
-    )
+    const material = decorateCryptographicMaterial({ suite } as any)
     const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
     material.setUnencryptedDataKey(dataKey, {
```

Comment:
Same as above: shouldn't this keyring trace (second argument) be deleted entirely?